keras_core/export/export_lib_test.py

```diff
@@ -0,0 +1,594 @@
+            ],
+        )
+        export_archive.write_out(temp_filepath)
+        revived_layer = tf.saved_model.load(temp_filepath)
+        self.assertAllClose(
+            ref_output, revived_layer.call(ref_input).numpy(), atol=1e-6
+        )
+
+    def test_multi_input_output_functional_model(self):
+        temp_filepath = os.path.join(self.get_temp_dir(), "exported_model")
+        x1 = layers.Input((2,))
+        x2 = layers.Input((2,))
+        y1 = layers.Dense(3)(x1)
+        y2 = layers.Dense(3)(x2)
+        model = models.Model([x1, x2], [y1, y2])
+
+        ref_inputs = [tf.random.normal((3, 2)), tf.random.normal((3, 2))]
+        ref_outputs = model(ref_inputs)
+
+        export_archive = export_lib.ExportArchive()
+        export_archive.track(model)
+        export_archive.add_endpoint(
+            "serve",
+            model.call,
+            input_signature=[
+                [
+                    tf.TensorSpec(
+                        shape=(None, 2),
+                        dtype=tf.float32,
+                    ),
+                    tf.TensorSpec(
+                        shape=(None, 2),
+                        dtype=tf.float32,
+                    ),
+                ]
+            ],
+        )
+        export_archive.write_out(temp_filepath)
+        revived_model = tf.saved_model.load(temp_filepath)
+        self.assertAllClose(
+            ref_outputs[0].numpy(),
+            revived_model.serve(ref_inputs)[0].numpy(),
+            atol=1e-6,
+        )
+        self.assertAllClose(
+            ref_outputs[1].numpy(),
+            revived_model.serve(ref_inputs)[1].numpy(),
+            atol=1e-6,
+        )
+
+        # Now test dict inputs
+        model = models.Model({"x1": x1, "x2": x2}, [y1, y2])
+
+        ref_inputs = {
+            "x1": tf.random.normal((3, 2)),
+            "x2": tf.random.normal((3, 2)),
+        }
+        ref_outputs = model(ref_inputs)
+
+        export_archive = export_lib.ExportArchive()
+        export_archive.track(model)
+        export_archive.add_endpoint(
+            "serve",
+            model.call,
+            input_signature=[
+                {
+                    "x1": tf.TensorSpec(
+                        shape=(None, 2),
+                        dtype=tf.float32,
+                    ),
+                    "x2": tf.TensorSpec(
+                        shape=(None, 2),
+                        dtype=tf.float32,
+                    ),
+                }
+            ],
+        )
+        export_archive.write_out(temp_filepath)
+        revived_model = tf.saved_model.load(temp_filepath)
+        self.assertAllClose(
+            ref_outputs[0].numpy(),
+            revived_model.serve(ref_inputs)[0].numpy(),
+            atol=1e-6,
+        )
+        self.assertAllClose(
+            ref_outputs[1].numpy(),
+            revived_model.serve(ref_inputs)[1].numpy(),
+            atol=1e-6,
+        )
+
+    # def test_model_with_lookup_table(self):
```

Comment:
The issue was calling a model with a TextVectorization Layer with string tensors (as we try to standardize them to float upon model call): https://chat.google.com/room/AAAADD9-Dbs/mZnA6hEpffQ

Since JAX and Torch do not support string tensors, we're trying a workaround of calling the layer first as Matt suggested (but there are some strange SavedModel tracked resource errors that need to be resolved with this workaround -- we can revisit this in a separate PR).

Reply:
^ Disregard that

Thanks for the workaround using the Input Layer, I've added the test back, changed some of the Lookup layer export logic and it works now!

Reply:
Interesting, running into another one of those errors where a test fails in CI but passes using pytest directly on it.